internal/resources/gitrepository/resource_git_repository_test.go

```diff
@@ -55,105 +58,178 @@ func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
 		ScopeHelperResources:        commonscope.NewScopeHelperResources(),
 		GitRepositoryDataSourceVar:  gitRepositoryDataSourceVar,
 		GitRepositoryDataSourceName: fmt.Sprintf("data.%s.%s", ResourceName, gitRepositoryDataSourceVar),
+		Namespace:                   "tanzu-continuousdelivery-resources",
 	}
 }
 
+func getConfigureContextFunc() func(_ context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	if _, found := os.LookupEnv("ENABLE_GITREPO_ENV_TEST"); !found {
+		return authctx.ProviderConfigureContextWithDefaultTransportForTesting
+	}
+
+	return authctx.ProviderConfigureContext
+}
+
+func getSetupConfig(config *authctx.TanzuContext) error {
+	if _, found := os.LookupEnv("ENABLE_GITREPO_ENV_TEST"); !found {
+		return config.SetupWithDefaultTransportForTesting()
+	}
+
+	return config.Setup()
+}
+
 func TestAcceptanceForGitRepositoryResource(t *testing.T) {
 	testConfig := testGetDefaultAcceptanceConfig(t)
 
+	// If the flag to execute git repository tests is not found, run this as a mock test by setting up an http intercept for each endpoint.
+	_, found := os.LookupEnv("ENABLE_GITREPO_ENV_TEST")
+	if !found {
+		os.Setenv("TF_ACC", "true")
+		os.Setenv("TMC_ENDPOINT", "dummy.tmc.mock.vmware.com")
+		os.Setenv("VMW_CLOUD_API_TOKEN", "dummy")
+		os.Setenv("VMW_CLOUD_ENDPOINT", "console.cloud.vmware.com")
+
+		log.Println("Setting up the mock endpoints...")
+		testConfig.setupHTTPMocks(t)
+	} else {
+		// Environment variables with non default values required for a successful call to MKP
```

Comment:
Please run linters manually before sending out for review as these checkins are to feature branch and not main. lint: full stop at the end,

Reply:
I've tried but didn't get any linter issues might need to bump up the version.
Updated PTAL